nutils/topology.py

```diff
@@ -878,15 +932,15 @@ class Point(Topology):
 
   @types.aspreprocessor
   @types.apply_annotations
-  def _preprocess_init(self, trans:transform.stricttransform, opposite:transform.stricttransform=None):
+  def _preprocess_init(self, trans:types.tuple[transform.stricttransform], opposite:types.tuple[transform.stricttransform]=None):
     return (self, trans, trans if opposite is None else opposite), {}
 
   @_preprocess_init
   def __init__(self, trans, opposite):
     assert trans[-1].fromdims == 0
     references = elementseq.asreferences([element.getsimplex(0)], 0)
-    transforms = transformseq.PlainTransforms((trans,), 0)
-    opposites = transforms if opposite is None else transformseq.PlainTransforms((opposite,), 0)
+    transforms = functools.reduce(transformseq.ProductTransforms, [transformseq.PlainTransforms((transi,), 0) for transi in trans])
+    opposites = functools.reduce(transformseq.ProductTransforms, [transformseq.PlainTransforms((oppi,), 0) for oppi in opposite])
     super().__init__(references, transforms, opposites)
 
 class StructuredLine(Topology):
```

Comment:
Is this currently used? If not, won't it be better to remove it with the aim to transition StructuredTopology into 1D only?

Reply:
Removed.